crates/ark/src/variables/variable.rs

```diff
@@ -918,132 +931,216 @@ impl PositronVariable {
-                                    object: RObject::view(VECTOR_ELT(*object, index)),
-                                }
-                            },
+            // Actual promises, i.e. unevaluated promises can't be
+            // expanded in the variables pane so we would not get here.
 
-                            LISTSXP => {
-                                let mut pairlist = *object;
-                                let index = path_element.parse::<isize>().unwrap();
-                                for _i in 0..index {
-                                    pairlist = CDR(pairlist);
-                                }
-                                EnvironmentVariableNode::Concrete {
-                                    object: RObject::view(CAR(pairlist)),
-                                }
-                            },
+            let value = unsafe { PRVALUE(x) };
+            if r_is_unbound(value) {
+                x = unsafe { PRCODE(x) };
+            } else {
+                x = value;
+            }
+        }
 
-                            LGLSXP | RAWSXP | STRSXP | INTSXP | REALSXP | CPLXSXP => {
-                                if r_is_matrix(*object) {
-                                    EnvironmentVariableNode::Matrixcolumn {
-                                        object,
-                                        index: path_element.parse::<isize>().unwrap(),
-                                    }
-                                } else {
-                                    EnvironmentVariableNode::VectorElement {
-                                        object,
-                                        index: path_element.parse::<isize>().unwrap(),
-                                    }
-                                }
-                            },
+        Ok(EnvironmentVariableNode::Concrete {
+            object: RObject::view(x),
+        })
+    }
 
-                            _ => {
-                                return Err(harp::error::Error::InspectError { path: path.clone() })
-                            },
-                        }
-                    }
-                },
+    fn get_concrete_child_node(
+        object: RObject,
+        access_key: &String,
+    ) -> harp::Result<EnvironmentVariableNode> {
+        // Concrete nodes are objects that are treated as is. Accessing an element from them
+        // might result in special node types.
+
+        // First try to get child using a generic method
+        // When building the children list of nodes that use a custom `get_children` method, the access_key is
+        // formatted as "custom-{index}-{length(name)}-{name}". If the access_key has this format, we call the custom `get_child_at`,
+        // method, if there's one available:
+        let result = local!({
+            let parsed_access_key: Vec<&str> = access_key.splitn(4, '-').collect();
+
+            if parsed_access_key.len() != 4 {
+                return Ok(None);
```

Comment:
Should that be logged? Same question for other failure modes.

Reply:
Or I guess: Should that be an `Err()`?

Reply:
In this first specific case no, because that just means that the access key does not come from a 'custom' inspect method.

The others could be logged, but in theory they could still be false positives (even though unlikely), an object could have an access key that's `custom-key-a-b` and get parsed until the below, so my hunch is that we don't need to do anything here.